src/ChatRoom.tsx

```diff
@@ -0,0 +1,55 @@
+import { useCallback, useRef, useState } from 'react';
+import { Chat } from './Interfaces';
+import chatData from './chatData.json';
+import userData from './userData.json';
+import UserList from './UserList';
+import ChatList from './ChatList';
+import styled from 'styled-components';
+import ChatInput from './ChatInput';
+
+const Wrapper = styled.div`
+  font-size: 12px;
+  display: flex;
+  flex-direction: column;
+  width: 350px;
+  height: 100vh;
+  box-shadow: rgba(0, 0, 0, 0.15) 0 1px 20px;
+  border-radius: 20px;
+  margin: 0 auto;
+`;
+
+function ChatRoom() {
+  const [chats, setChats] = useState<Chat[]>(chatData.chats);
+  const [userId, setUserId] = useState(0);
+  const nextChatId = useRef(chatData.chats.length + 1);
+  const users = userData.users;
+
+  const addChat = useCallback(
+    (text: string) => {
+      const chat = {
+        id: nextChatId.current,
+        userId: userId,
+        text,
+        date: String(new Date()),
+      };
+      setChats(chats.concat(chat));
```

Comment:
```suggestion
      setChats([...chats, chat]);
```
효신님 답변에 이어!! spread를 사용하면 요렇게 수정하면 됩니다 ㅎㅎ